src/file.cr

```diff
@@ -36,6 +36,27 @@ class File < IO::FileDescriptor
     super(fd, blocking)
   end
 
+  # Opens the file named by *filename*.
+  #
+  # *mode* must be one of the following file open modes:
+  # ```text
+  # Mode | Description
+  # -----+------------------------------------------------------
+  # r    | Read-only, starts at the beginning of the file.
+  # r+   | Read-write, starts at the beginning of the file.
+  # w    | Write-only, truncates existing file to zero length or
+  #      | creates a new file if the file doesn't exists.
+  # w+   | Read-write, truncates existing file to zero length or
+  #      | creates a new file if the file doesn't exists.
+  # a    | Write-only, starts at the end of the file,
+  #      | creates a new file if the file doesn't exists.
+  # a+   | Read-write, starts at the end of the file,
+  #      | creates a new file if the file doesn't exists.
+  # rb   | Same as the 'r' mode but in binary file mode.
+  # wb   | Same as the 'w' mode but in binary file mode.
+  # ab   | Same as the 'a' mode but in binary file mode.
+  # ```
+  # In binary file mode, line endings are not converted to CRLF on Windows.
```

Comment:
Binary mode treats files as binary data. Convertions of line endings is just an example of that (although it's probably the sole significant consequence).